pkg/common/utils/utils.go

```diff
@@ -238,7 +241,27 @@ func MergeMaps(first, second map[string]string) map[string]string {
 }
 
 // find user name from pod label
-func GetUserFromPod(pod *v1.Pod) string {
+func GetUserFromPod(pod *v1.Pod) (string, []string) {
+	if pod.Annotations[userInfoKey] != "" {
+		userInfoJSON := pod.Annotations[userInfoKey]
+		var userGroup si.UserGroupInformation
+		err := json.Unmarshal([]byte(userInfoJSON), &userGroup)
+		if err != nil {
+			log.Logger().Error("unable to process user info annotation", zap.Error(err))
+			return constants.DefaultUser, nil
+		}
+		user := userGroup.User
+		groups := userGroup.Groups
```

Comment:
New to the project so this may be a useless suggestion. Do you care if groups are empty? Should there be logic for a default group? Or at least a warning if groups are empty?

Reply:
Makes sense, I added a default group + logging

Reply:
I disagree that we should have default groups, as this creates "magic" values that behave in somewhat unpredictable ways. An empty group list is allowed explicitly by the design, so we should not add additional groups here.